code/datums/cinematic.dm

```diff
@@ -38,49 +36,48 @@ GLOBAL_LIST_EMPTY(cinematics)
 	var/stop_ooc = TRUE //Turns off ooc when played globally.
 
 /datum/cinematic/New()
-	GLOB.cinematics += src
 	screen = new(src)
 
 /datum/cinematic/Destroy()
-	GLOB.cinematics -= src
+	for(var/CC in watching)
+		if(!CC)
+			continue
+		var/client/C = CC
+		C.screen -= screen
 	QDEL_NULL(screen)
-	for(var/mob/M in locked)
+	QDEL_NULL(special_callback)
+	for(var/MM in locked)
+		if(!MM)
+			continue
+		var/mob/M = MM
 		M.notransform = FALSE
 	return ..()
 
 /datum/cinematic/proc/play(watchers)
-	//Check if you can actually play it (stop mob cinematics for global ones) and create screen objects
-	for(var/A in GLOB.cinematics)
-		var/datum/cinematic/C = A
-		if(C == src)
-			continue
-		if(C.is_global || !is_global)
-			return //Can't play two global or local cinematics at the same time
+	//Check if you can actually play it (stop mob cinematics for global ones)
+	if(SEND_GLOBAL_SIGNAL(COMSIG_GLOB_PLAY_CINEMATIC, src) & COMPONENT_GLOB_BLOCK_CINEMATIC)
+		return
 
-	//Close all open windows if global
-	if(is_global)
-		SStgui.close_all_uis()
+	//We are now playing this cinematic
+
+	//Handle what happens when a different cinematic tries to play over us
+	RegisterSignal(SSdcs, COMSIG_GLOB_PLAY_CINEMATIC, .proc/replacement_cinematic)
 
 	//Pause OOC
 	var/ooc_toggled = FALSE
 	if(is_global && stop_ooc && GLOB.ooc_allowed)
 		ooc_toggled = TRUE
 		toggle_ooc(FALSE)
 
-
-	for(var/mob/M in GLOB.mob_list)
-		if(M in watchers)
-			M.notransform = TRUE //Should this be done for non-global cinematics or even at all ?
-			locked += M
-			//Close watcher ui's
-			SStgui.close_user_uis(M)
-			if(M.client)
-				watching += M.client
-				M.client.screen += screen
-		else
-			if(is_global)
-				M.notransform = TRUE
-				locked += M
+	for(var/MM in watchers)
+		var/mob/M = MM
+		M.notransform = TRUE //Should this be done for non-global cinematics or even at all ?
+		locked += M
+		//Close watcher ui's
+		SStgui.close_user_uis(M)
+		if(M.client)
+			watching += M.client
+			M.client.screen += screen
 
```

Comment:
If there's no client you may want to listen for if a client joins mid cinematic

Reply:
Good idea!